packages/template-retail-react-app/app/components/item-variant/item-price.jsx

```diff
@@ -43,74 +40,66 @@ PricePerItem.propTypes = {
  */
 const ItemPrice = ({currency, align = 'right', baseDirection = 'column', ...props}) => {
     const variant = useItemVariant()
-    const {data: basket} = useCurrentBasket()
-    const {currency: activeCurrency} = useCurrency()
-    const intl = useIntl()
-
-    const {price, basePrice, priceAfterItemDiscount} = variant
-    const isProductASet = variant?.type?.set
-
-    const displayPrice =
-        typeof priceAfterItemDiscount === 'number' ? Math.min(price, priceAfterItemDiscount) : price
-
-    const hasDiscount = displayPrice !== price
-
+    const {price, priceAfterItemDiscount} = variant
+    const isASet = variant?.type?.set
+    const isMaster = variant?.type?.master
+    let priceData
+    // When variant has basket pricing, we should prioritize these prices for displaying
+    // since they may have take promotion/discount into account
+    // NOTE: try NOT to re-calculate these values since basket-level discount is complicated
+    if (variant?.itemId) {
+        priceData = {
+            currentPrice: priceAfterItemDiscount,
+            listPrice: undefined,
+            pricePerUnit: variant?.pricePerUnit,
+            isASet,
+            isMaster,
+            isRange: isASet || isMaster || false,
+            isOnSale: price > priceAfterItemDiscount
+        }
+    } else {
+        // for wishlist page we extract price info from variant/product obj
+        priceData = getPriceData(variant)
+    }
+    const isDesktop = useBreakpointValue({base: false, lg: true})
+    const isOnSale = price > priceAfterItemDiscount || priceData?.isOnSale
     return (
         <Stack
             textAlign={align}
-            direction={hasDiscount ? 'column' : {base: baseDirection, lg: 'row'}}
+            direction="column"
             justifyContent={align === 'left' ? 'flex-start' : 'flex-end'}
             alignItems="baseline"
-            spacing={hasDiscount ? 0 : 1}
+            spacing={isOnSale ? 0 : 1}
             wrap="nowrap"
             {...props}
         >
-            {basePrice && price !== basePrice && (
-                <HideOnDesktop>
-                    <PricePerItem currency={currency} basePrice={basePrice} basket={basket} />
-                </HideOnDesktop>
+            {variant?.quantity > 1 && !isASet && priceData?.pricePerUnit && !isDesktop && (
```

Comment:
How about showing the `!isDesktop` first? So it's clear what the difference is between here and on line 94 below. One is for mobile, and the other is for desktop.